tests/Endpoints/SearchTest.php

```diff
@@ -24,69 +24,59 @@ public function testBasicSearch(): void
     {
         $response = $this->index->search('prince');
 
-        $this->assertArrayHasKey('hits', $response->toArray());
-        $this->assertArrayHasKey('offset', $response->toArray());
-        $this->assertArrayHasKey('limit', $response->toArray());
-        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
-        $this->assertArrayHasKey('query', $response->toArray());
+        $this->assertEstimatedPagination($response->toArray());
         $this->assertSame(2, $response->getEstimatedTotalHits());
         $this->assertCount(2, $response->getHits());
 
         $response = $this->index->search('prince', [], [
             'raw' => true,
         ]);
 
-        $this->assertArrayHasKey('hits', $response);
-        $this->assertArrayHasKey('offset', $response);
-        $this->assertArrayHasKey('limit', $response);
-        $this->assertArrayHasKey('processingTimeMs', $response);
-        $this->assertArrayHasKey('query', $response);
+        $this->assertEstimatedPagination($response);
         $this->assertSame(2, $response['estimatedTotalHits']);
     }
 
+    public function testBasicSearchWithFinitePagination(): void
```

Comment:
we should really decide to call it either numbered pagination or finite pagination or page selection pagination 😂
I'm saying this for my scope as well

Reply:
Yes, I agree with you! Maybe we can check with docs, what they choose...